prog/weaver/main.go

```diff
@@ -68,7 +76,7 @@ func checkForUpdates() {
 	}
 	resp, err := checkpoint.Check(&params)
 	handleResponse(resp, err)
-	checkpoint.CheckInterval(&params, updateCheckPeriod, handleResponse)
+	checker = checkpoint.CheckInterval(&params, updateCheckPeriod, handleResponse)
```

Comment:
As discussed, we should change `CheckInterval` s.t. it performs a check immediately. That way we don't need the `Check` call, can get rid of the intermediate vars for `handleResponse` and `params`, and, most importantly, don't need to run this function in its own goroutine, thus ensuring that `checker` will always be set, which in turn will simplify the status reporting code.